lib/spack/spack/solver/concretize.lp

```diff
@@ -459,7 +459,9 @@ version_weight(Package, Weight) :- external_version(Package, Version, Weight).
 version(Package, Version) :- external_version(Package, Version, Weight).
 
 % if a package is not buildable (external_only), only externals are allowed
-external(Package) :- external_only(Package), node(Package).
+external(Package) :- external_only(Package),
```

Comment:
I think this will allow you to reuse a non-external even when `buildable: false` for that package. Can you add a test for this?

Reply:
I don't think this is wrong. It will allow what you say only if the non-external is already installed, so not buildable, but built.

Disallowing that case causes the bug in this way:
1. Install an external that is not buildable
2. Can't reuse the same external at the next concretization